addons/attached_objects/functions/fnc_attachBone.sqf

```diff
@@ -0,0 +1,83 @@
+#include "script_component.hpp"
+/*
+ * Author: Ampersand
+ * Attaches the given objects to a parent object selected by Zeus.
+ *
+ * Arguments:
+ * 0: Objects <ARRAY>
+ * 1: Is Bone <BOOLEAN>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_object] call zen_attached_objects_fnc_attachBone
+ *
+ * Public: No
+ */
+
+[_this, {
+    params ["_successful", "_objects"];
+
+    if (!_successful) exitWith {};
+
+    curatorMouseOver params ["_type", "_entity"];
+    if (_type != "OBJECT") exitWith {};
+
+    // Prevent attaching object to itself
+    private _index = _objects find _entity;
+
+    if (_index != -1 && {count _objects == 1}) exitWith {
+        [LSTRING(CannotAttachToSelf)] call EFUNC(common,showMessage);
+    };
+
+    _objects deleteAt _index;
+
+    if (isNil QGVAR(modelSelections)) then {
+        GVAR(modelSelections) = createHashMap;
+    };
+
+    private _selections = GVAR(modelSelections) getOrDefaultCall [getText (configOf _entity >> "model"), {
+            [""] + (_entity selectionNames LOD_MEMORY select {
+                _entity selectionVectorDirAndUp [_x, LOD_MEMORY] isNotEqualTo [[0,0,0], [0,0,0]]
+            })
+    }, true];
+
+    [LSTRING(AttachTo), [
+        [
+            "COMBO",
+            "Selection",
+            [_selections, _selections, _selections find _selection]
+        ],
+        [
+            "TOOLBOX",
+            ["Orientation", "Selection: match selection orientation (i.e. along gun barrel). Relative: maintain current orientation relative to selection."],
+            [true, 1, 2, ["Selection", "Relative"]],
+            true
+        ]
+    ], {
+        params ["_values", "_args"];
+        _values params ["_selection", "_isRelative"];
+        _args params ["_objects", "_entity"];
+
+        {
+            if (_selection == "") then {
+                private _dirAndUp = [_entity vectorWorldToModel vectorDir _x, _entity vectorWorldToModel vectorUp _x];
+                _x attachTo [_entity];
+                [QEGVAR(common,setVectorDirAndUp), [_x, _dirAndUp], _x] call CBA_fnc_targetEvent;
+            } else {
+                [_x, _entity, _selection, _isRelative] call FUNC(attachToSelection);
+            };
+        } forEach _objects params ["_selectionPos", "_selectionY", "_selectionZ"];
```

Comment:
Is this actually the intention ?

^ Since `_objects` will be iterated before `params` gets to it, and `params` returns bool based on the success of, or use of a default value in, the creation of the mentioned vars, so even putting it as `forEach (...)` would give an error, IIUC, and the mentioned vars are only used after the `forEach`.

Reply:
Ah, yeah you're right. `forEach` does return the last iteration, but it's not a good use for it, and it was in the wrong place relative to the if anyway.